src/pytest_cov/plugin.py

```diff
@@ -263,20 +263,24 @@ def pytest_terminal_summary(self, terminalreporter):
 
         terminalreporter.write('\n' + self.cov_report.getvalue() + '\n')
 
-        if self.options.cov_fail_under is not None and self.options.cov_fail_under > 0:
-            if self.cov_total < self.options.cov_fail_under:
+        fail_under = self.options.cov_fail_under
+        if fail_under is not None and fail_under > 0:
+            str_fail_under = str(
+                round(fail_under, 2) if fail_under % 1 else int(fail_under)
```

Comment:
I wonder if using a Decimal would simplify this.

Reply:
this means that fail_under `--cov-fail-under==1.0` fails also?

Reply:
Ah nevermind, now I get what it does. It would display 0.0 and 0 as "0". But that seems pointless. Why not just have ` ... %s ... " % (round(self.options.cov_fail_under, 2) self.cov_total))`? Clear and simple.